From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Thu, 13 Jul 2023 19:54:12 +0200
Subject: [PATCH] Modifiable Bukkit enums - Inject runtime versions - Material
 - Register key for Fiddle values

Registers the namespaced keys of Fiddle `Material` instances so that they can be returned by `Material.getKey()`,
by calling the corresponding method added by the "Modifiable Bukkit enums - Inject runtime versions - Material - Get key for Fiddle values" patch in the `fiddle-api` module.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index 7b932fcee235acf2e7ee56003b90c84e7136c146..28c28156f130360953418baa5d2327aa34f62bb6 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -251,6 +251,11 @@ public class Main {
                 }).build())
                 .make()
                 .load(ClassLoader.getSystemClassLoader(), net.bytebuddy.dynamic.loading.ClassLoadingStrategy.Default.INJECTION);
+            // Fiddle start - modifiable Bukkit enums - inject runtime versions - Material - keys for Fiddle values
+            for (var fiddleEnumName : fiddleEnumNames) {
+                org.fiddlemc.fiddle.material.MaterialNameAndKeyConversion.registerFiddleEnumNameToNamespacedKey(fiddleEnumName.left(), fiddleEnumName.right().getNamespace(), fiddleEnumName.right().getPath());
+            }
+            // Fiddle end - modifiable Bukkit enums - inject runtime versions - Material - keys for Fiddle values
             // Fiddle end - modifiable Bukkit enums - inject runtime versions - Material
 
             dedicatedserversettings.forceSave();
